src/user/room.ts

```diff
@@ -425,63 +425,106 @@ export class Room extends Accessory implements Sayable {
-    replyToList = replyToList.concat(mention || [])
-
-    const mentionAliasList = await Promise.all(
-                                      replyToList.map(
-                                        async c => await this.alias(c) || c.name()
-                                      )
-                                    )
-
     log.verbose('Room', 'say(%s, %s)',
-                        textOrContactOrFileOrUrl,
-                        mentionAliasList.join(', '),
+                        textOrListOrContactOrFileOrUrl,
+                        mentionList.join(', '),
                 )
 
     let text: string
 
-    if (typeof textOrContactOrFileOrUrl === 'string') {
+    if (typeof textOrListOrContactOrFileOrUrl === 'string') {
 
-      if (mentionAliasList.length > 0) {
-        // const AT_SEPRATOR = String.fromCharCode(8197)
-        const AT_SEPRATOR = FOUR_PER_EM_SPACE
-        const mentionList = mentionAliasList.map(roomAlias => '@' + roomAlias).join(AT_SEPRATOR)
+      if (mentionList.length > 0) {
+        const AT_SEPARATOR = FOUR_PER_EM_SPACE
+        const mentionAlias = await Promise.all(mentionList.map(async contact =>
+          '@' + (await this.alias(contact) || contact.name())
+        ))
+        const mentionText = mentionAlias.join(AT_SEPARATOR)
 
-        text = mentionList + ' ' + textOrContactOrFileOrUrl
+        text = mentionText + ' ' + textOrListOrContactOrFileOrUrl
       } else {
-        text = textOrContactOrFileOrUrl
+        text = textOrListOrContactOrFileOrUrl
       }
       const receiver = {
-        contactId : replyToList.length && replyToList[0].id || undefined,
+        contactId : mentionList.length && mentionList[0].id || undefined,
         roomId    : this.id,
       }
       await this.puppet.messageSendText(
         receiver,
         text,
-        replyToList.map(c => c.id),
+        mentionList.map(c => c.id),
       )
-    } else if (textOrContactOrFileOrUrl instanceof FileBox) {
+    } else if (textOrListOrContactOrFileOrUrl instanceof FileBox) {
+      /**
+       * 2. File Message
+       */
       await this.puppet.messageSendFile({
         roomId: this.id,
-      }, textOrContactOrFileOrUrl)
-    } else if (textOrContactOrFileOrUrl instanceof Contact) {
+      }, textOrListOrContactOrFileOrUrl)
+    } else if (textOrListOrContactOrFileOrUrl instanceof Contact) {
+      /**
+       * 3. Contact Card
+       */
       await this.puppet.messageSendContact({
         roomId: this.id,
-      }, textOrContactOrFileOrUrl.id)
-    } else if (textOrContactOrFileOrUrl instanceof UrlLink) {
+      }, textOrListOrContactOrFileOrUrl.id)
+    } else if (textOrListOrContactOrFileOrUrl instanceof UrlLink) {
       /**
        * 4. Link Message
        */
       await this.puppet.messageSendUrl({
         contactId : this.id
-      }, textOrContactOrFileOrUrl.payload)
+      }, textOrListOrContactOrFileOrUrl.payload)
+    } else if (textOrListOrContactOrFileOrUrl instanceof Array) {
```

Comment:
Please add a sub-function for the following codes for `sayTemplateStringsArray()`.

```ts
    } else if (textOrListOrContactOrFileOrUrl instanceof Array) {
      return sayTemplateStringsArray(
        textOrListOrContactOrFileOrUrl, 
        ...mentionList,
      )
    }
```